osu.Game/Rulesets/Mods/ModJudgementChallenge.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Rulesets.Judgements;
+using osu.Framework.Bindables;
+using osu.Game.Rulesets.Scoring;
+using osu.Game.Configuration;
+using osu.Game.Overlays.Settings;
+
+namespace osu.Game.Rulesets.Mods
+{
+    public abstract class ModJudgementChallenge : ModChallenge
+    {
+        public override string Name => "Judgement Challenge";
+        public override string Acronym => "JC";
+        public override IconUsage? Icon => FontAwesome.Solid.Gavel;
+        public override string Description => "Fail the beatmap if your judgements go above specified values.";
+
+        [SettingSource("Maximum non-best judgements", "Maximum number of non-best judgements before fail (disregards judgement type).", 1, SettingControlType = typeof(SettingsNumberBox))]
+        public Bindable<int?> MaxNonBest { get; } = new Bindable<int?>
+        {
+            Default = null,
+            Value = null
+        };
```

Comment:
I'm not really sure about textbox instead of slider here. The range probably doesn't have to be too large, maybe 0-25 (or 50) is enough.

Also I don't know about the UX of this. You probably want to split this to a control that switches between inputting the number of non-best judgements, or the number of judgements for each type, i.e. have a custom setting bindable of a struct type and a custom control that maps the value of that struct to one slider or many sliders based on a checkbox, something like that).

Reply:
You could temporarily (or even permanently?) avoid disabling the slider by considering the maximum value to be for allowing judgement type regardless of count (display as "no maximum" or just "all").

You can override the tooltip display of that "maximum value" with a custom slider (see [`ModMuted`](https://github.com/ppy/osu/blob/2e21d75b101d4aabb3a2dba05ef2e7306c6976ce/osu.Game/Rulesets/Mods/ModMuted.cs#L111-L114)).

Reply:
Didn't know you could do that... seems like a clean enough solution.